packages/contracts/contracts/voting/majority/IMajorityVoting.sol

```diff
@@ -17,126 +17,132 @@ interface IMajorityVoting {
+        uint256 totalVotingPower;
         mapping(address => VoteOption) voters;
         IDAO.Action[] actions;
     }
 
-    /// @notice Emitted when a vote is created.
-    /// @param voteId  The ID of the vote.
-    /// @param creator  The creator of the vote.
-    /// @param metadata The IPFS hash pointing to the proposal metadata.
-    event VoteCreated(uint256 indexed voteId, address indexed creator, bytes metadata);
-
-    /// @notice Emitted when a vote is casted by a voter.
-    /// @param voteId The ID of the vote.
+    /// @notice Emitted when a vote is cast by a voter.
+    /// @param proposalId The ID of the proposal.
     /// @param voter The voter casting the vote.
     /// @param choice The vote option chosen.
-    /// @param voteWeight The weight of the casted vote.
-    event VoteCast(uint256 indexed voteId, address indexed voter, uint8 choice, uint256 voteWeight);
+    /// @param votingPower The voting power behind this vote.
+    event VoteCast(
+        uint256 indexed proposalId,
+        address indexed voter,
+        uint8 choice,
+        uint256 votingPower
+    );
+
+    /// @notice Emitted when a proposal is created.
+    /// @param proposalId The ID of the proposal.
+    /// @param creator  The creator of the proposal.
+    /// @param metadata The IPFS hash pointing to the proposal metadata.
+    event ProposalCreated(uint256 indexed proposalId, address indexed creator, bytes metadata);
 
-    /// @notice Emitted when a vote is executed.
-    /// @param voteId The ID of the vote.
-    /// @param execResults The bytes array resulting from the vote execution in the associated DAO.
-    event VoteExecuted(uint256 indexed voteId, bytes[] execResults);
+    /// @notice Emitted when a proposal is executed.
+    /// @param proposalId The ID of the proposal.
+    /// @param execResults The bytes array resulting from the proposal execution in the associated DAO.
+    event ProposalExecuted(uint256 indexed proposalId, bytes[] execResults);
 
-    /// @notice Emitted when the vote configuration is updated.
-    /// @param relativeSupportThresholdPct The relative support threshold in percent.
+    /// @notice Emitted when the vote settings are updated.
+    /// @param relativeSupportThresholdPct The support threshold in percent.
     /// @param totalSupportThresholdPct The total support threshold in percent.
     /// @param minDuration The minimal duration of a vote.
-    event ConfigUpdated(
+    event VoteSettingsUpdated(
         uint64 totalSupportThresholdPct,
         uint64 relativeSupportThresholdPct,
         uint64 minDuration
     );
 
-    /// @notice Sets the vote configuration.
+    /// @notice Changes the vote settings.
     /// @param _totalSupportThresholdPct The total support threshold in percent.
     /// @param _relativeSupportThresholdPct The relative support threshold in percent.
     /// @param _minDuration The minimal duration of a vote.
-    function setConfiguration(
+    function changeVoteSettings(
```

Comment:
Another one here, if we are changing the name from Vote to Proposal this should be `changeProposalSettings`

Reply:
On second thought: these are really the vote settings.
In the next step, we will probably make a struct

`struct VoteSettings`
containing the parameters that is different from `struct Proposal`.

`MajorityVoting` can contain `VoteSettings` even if we call it a proposal.

Reply:
see https://github.com/aragon/core/pull/146#discussion_r1031554135